src/controller/invoice-controller.ts

```diff
@@ -225,9 +227,14 @@ export default class InvoiceController extends BaseController {
         return;
       }
 
-      const invoice: Invoice = await InvoiceService.createInvoice(params);
+      const invoice: Invoice = await AppDataSource.manager.transaction(async (manager) =>
+        Promise.resolve(new InvoiceService(manager).createInvoice(params)));
```

Comment:
You can just do `return new InvoiceService(manager).createInvoice(params));` here (without the `Promise.resolve`), because your function is asynchronous. Saves another wrapper.